File: python/백준/그래프/프로이드-워셜/11404 플로이드.py
import sys

input = sys.stdin.readline

n = int(input())
m = int(input())

INF = int(1e9)
graph = [[INF] * (n + 1) for _ in range(n + 1)]

for a in range(1, n + 1):
    for b in range(1, n + 1):
        if a == b:
            graph[a][b] = 0
    
for _ in range(m):
    a, b, c = map(int, input().split())
    
    graph[a][b] = min(graph[a][b], c)

for k in range(1, n + 1):
    for a in range(1, n +1):
        for b in range(1, n + 1):
            graph[a][b] = min(graph[a][k] + graph[k][b], graph[a][b])
            
            
for a in range(1, n + 1):
    for b in range(1, n + 1):
        if graph[a][b] == INF:
            print(0, end=' ')
        else:
            print(graph[a][b], end=" ")
    print()

File: cpp/백준/그래프/프로이드-워셜/11404 플로이드.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

const int INF = 1e9;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int n, m;
    cin >> n;
    cin >> m;

    vector<vector<int>> graph(n + 1, vector<int>(n + 1, INF));
    for (int a = 1; a <= n; ++a) graph[a][a] = 0;

    for (int i = 0; i < m; ++i) {
        int a, b, c;
        cin >> a >> b >> c;
        graph[a][b] = min(graph[a][b], c);
    }

    for (int k = 1; k <= n; ++k) {
        for (int a = 1; a <= n; ++a) {
            for (int b = 1; b <= n; ++b) {
                graph[a][b] = min(graph[a][b], graph[a][k] + graph[k][b]);
            }
        }
    }

    for (int a = 1; a <= n; ++a) {
        for (int b = 1; b <= n; ++b) {
            if (graph[a][b] == INF) cout << 0;
            else cout << graph[a][b];
            if (b < n) cout << " ";
        }
        cout << "\n";
    }
    return 0;
}
